### ISSUE TITLE
Scattermat documentation incorrect

### ISSUE THREAD

**Issue by funnell** [JuliaStats/StatsBase.jl#219]
```
help?> scattermat
search: scattermat

  scattermat(X, [wv::WeightVec]; mean=nothing, vardim=1)

  Compute the scatter matrix, which is an unnormalized covariance matrix. A weighting vector wv can be specified to weight the estimate.

     Arguments
    ≡≡≡≡≡≡≡≡≡≡≡

    •    mean=nothing: a known mean value. nothing indicates that the mean is


  unknown, and the function will compute the mean. Specifying mean=0 indicates that the data are centered and hence there's no need to subtract the mean.

    •    vardim=1: the dimension along which the variables are organized.


  When vardim = 1, the variables are considered columns with observations in rows; when vardim = 2, variables are in rows with observations in columns.
```

```
julia> scattermat(randn(3,3), mean=zeros(3))
ERROR: function scattermat does not accept keyword arguments
 in kwfunc(::Any) at ./boot.jl:236
```

```
julia --version
julia version 0.5.0
```

Looking at cov.jl, the scattermat function definitions are different depending on the julia version being run.

**Comment by ararslan** [comment]
`scattermat` is defined in terms of `Base.covzm` for Julia < 0.5.0-dev+679, since `covzm` was removed from Base in that version. Now, why the actual `scattermat` function API is different before and after 0.5.0-dev+679 I'm not sure, but my guess would be that it was an oversight. I think @andreasnoack knows this code the best. Andreas, can you share any insight?

**Comment by andreasnoack** [comment]
This is actually fixed on master but see #442
